build.gradle.kts

```diff
@@ -174,37 +157,73 @@ subprojects {
 
 	// Source encoding
 	tasks {
+
+		withType<ManifestTask> {
+			outputs.file(File(buildDir, "resources/main/META-INF/MANIFEST.MF"))
+					.withPropertyName("outputFile")
+
+		}
 		// shadowjar & relocation
-		val relocateShadowJar by creating(com.github.jengelman.gradle.plugins.shadow.tasks.ConfigureShadowRelocation::class) {
-			target = shadowJar.get()
+		val relocateShadowJar by creating(ConfigureShadowRelocation::class) {
+			target = named<ShadowJar>("shadowJar").get()
```

Comment:
I would remove this change. Using the lazy `named<ShadowJar>("shadowJar")` is redundant if immediately call `.get()` on the provider.

Reply:
same reason as above, `shadowJar` property is not available if plugin is not applied

Reply:
ah makes sense then